WWST-1275: Aeotec Water Sensor 6 - ZW122-A

joshk-trility opened the pull request:
Updated generic water sensor DTH with support for the water sensor.
Added missing security encapsulation functions.

joshk-trility committed:
WWST-1275: Aeotec Water Sensor 6 - ZW122-A

Updated generic water sensor DTH with support for the water sensor.
Added missing security encapsulation functions.

joshk-trility commented:
@greens This PR supersedes the previous Aeotec water sensor one that @aclly posted. It should also address concerns you had from that PR.
Ready for review.

greens commented on devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy:
```diff
@@ -69,6 +84,16 @@ def updated() {
+		def commands = []
+		// Tell sensor to send us battery information instead of USB power information
+		commands << encap(zwave.configurationV1.configurationSet(parameterNumber: 0x5E, scaledConfigurationValue: 1, size: 1))
+		commands << encap(zwave.wakeUpV1.wakeUpIntervalSet(seconds: 4 * 3600, nodeid: zwaveHubNodeId))
```
I think 4 hours is the default we automatically configure wakeup devices to use.

greens replied:
The old code that did that was probably just stale.

joshk-trility replied:
Done.

greens commented on devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy:
```diff
@@ -53,8 +56,20 @@ metadata {
-	sendEvent(name: "checkInterval", value: (2 * 12 + 2) * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if (zwaveInfo.mfr.equals("0086"))
+		// 4 hour ping for Aeotec
+		sendEvent(name: "checkInterval", value: 4 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
typically the formula for the checkinterval is `(wakeupinterval)*2 + 2 minutes` which lets us miss one check-in and gives us some wiggle room before marking a device offline.

joshk-trility replied:
Done.

greens commented on devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy:
```diff
@@ -53,8 +56,20 @@ metadata {
+	response(cmds)
 }
 
 def installed() {
```
looks like you have two `installed()` methods now.

joshk-trility replied:
Done.

joshk-trility committed:
Fixes to water sensor DTH based on comments

Removed duplicate installed() method. Removed redundant wakeUpInterval
setting. Fixed checkInterval for Aeotec Water Sensor.
